plumbing/api.go

```diff
@@ -27,7 +27,7 @@ import (
 type API struct {
 	logger logging.EventLogger
 
-	chain        *chn.Reader
+	chain        chain.ReadStore
```

Comment:
even more idiomatic would be to define an interface giving the subset of chain.readstore we need here and have chain's type be that. consumers should generally define the interface they need. but also ok to leave as it given that this is consistent with the way the code works throughout gfc.

Reply:
Thanks. I opt to leave this one as is since it's an internal package. Ultimately, plumbing will be the ~only the consumer of this interface, so most methods should be required.